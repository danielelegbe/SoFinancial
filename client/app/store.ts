import { configureStore } from '@reduxjs/toolkit';
import userReducer from '../features/user/userSlice';
import chatReducer from '../features/chat/chatSlice';

export const store = configureStore({
  reducer: {
    user: userReducer,
    chat: chatReducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
