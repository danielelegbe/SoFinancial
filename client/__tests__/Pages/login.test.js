import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Login from '../../pages/login';
import { store } from '../../app/store';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';

describe('Login page tests', () => {
  afterEach(() => {
    cleanup()
  })

  test('test', () => {
    render(<Provider store={store}><Login /></Provider>)
    const login = screen.getByTestId('login');
    expect(login).toBeInTheDocument();
  })

  test('snapshot login page test', () => {
    const login = renderer.create(<Provider store={store}><Login /></Provider>).toJSON();
    expect(login).toMatchSnapshot()
  })
})
  