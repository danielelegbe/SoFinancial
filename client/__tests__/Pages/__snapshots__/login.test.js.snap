// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page tests snapshot login page test 1`] = `
<div
  className="css-ok4vqq"
  data-testid="login"
>
  <div
    className="chakra-stack css-v96ksb"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="forex icon"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="forex icon"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=%2Fforex.png&w=256&q=75"
          srcSet="/_next/image?url=%2Fforex.png&w=128&q=75 1x, /_next/image?url=%2Fforex.png&w=256&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </div>
    <p
      className="chakra-text css-6g8l6z"
    >
      Login
    </p>
    <div
      className="css-a6rva7"
    >
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="chakra-stack css-137fd6w"
        >
          <input
            className="chakra-input css-0"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Username"
            value=""
          />
          <input
            className="chakra-input css-0"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            className="chakra-button css-xpgcky"
            disabled={false}
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
